tests/vxlan/test_vnet_vxlan.py

```diff
@@ -0,0 +1,167 @@
+    """
+    @summary: Paramterized fixture that tests the Disabled, Enabled, and Cleanup configs for VxLAN
+    @param setup: Pytest fixture that provides access to minigraph facts
+    @param request: Contains the parameter (Disabled, Enabled, or Cleanup) for the current test iteration
+    @param duthost: DUT host object
+    @param returns: VxLAN status, and the test scenario
+    """
+    vxlan_enabled = False
+    if request.param == "Disabled":
+        vxlan_enabled = False
+    elif request.param == "Enabled":
+        mg_facts = setup
+
+        duthost.shell("sonic-clear fdb all")
+        
+        attach_to = mg_facts["minigraph_vlan_interfaces"][0]['attachto']
+        member_to_remove = mg_facts["minigraph_vlans"][attach_to]['members'][0]
+        duthost.shell("docker exec -i database redis-cli -n 4 del \"VLAN_MEMBER|{}|{}\"".format(attach_to, member_to_remove))
+
+        apply_dut_config_files(duthost) 
+
+        vxlan_enabled = True
+    elif request.param == "Cleanup" and vc.CLEANUP:
+        vxlan_enabled = True
+        render_template_to_host("vnet_routes.j2", duthost, vc.DUT_VNET_ROUTE_CONFIG, vc.VNET_CONFIG, op="DEL")
+        duthost.shell("docker cp {} swss:/vnet.route.json".format(vc.DUT_VNET_ROUTE_CONFIG))
+        duthost.shell("docker exec swss sh -c \"swssconfig /vnet.route.json\"")
+        sleep(3)
+        cleanup_dut_vnets(duthost, setup)
+        cleanup_vxlan_tunnels(duthost)
+    return vxlan_enabled, request.param
+
+
+def test_vnet_vxlan(setup, vxlan_status, duthost, ptfhost):
+    """
+    @summary: Test case for VNET VxLAN
+    @param setup: Pytest fixture that sets up PTF and DUT hosts and yields minigraph facts
+    @param vxlan_status: Parameterized pytest fixture used to test different VxLAN configurations
+    @param duthost: DUT host object
+    @param ptfhost: PTF host object
+    """
+    vxlan_enabled, scenario = vxlan_status
+
+    host_vars = duthost.host.options['variable_manager']._hostvars[duthost.hostname]
+    inventory = host_vars['inventory_file'].split('/')[-1]
+    secrets = duthost.host.options['variable_manager']._hostvars[duthost.hostname]['secret_group_vars']
+
+    logger.info("vxlan_enabled={}, scenario={}".format(vxlan_enabled, scenario))
+
+    log_file = "/tmp/vnet-vxlan.Vxlan.{}.{}.log".format(scenario, datetime.now().strftime('%Y-%m-%d-%H:%M:%S'))
+    ptf_params = {"vxlan_enabled": vxlan_enabled,
+                  "config_file": '/tmp/vnet.json',
+                  "sonic_admin_user": secrets[inventory]['sonicadmin_user'],
+                  "sonic_admin_password": secrets[inventory]['sonicadmin_password'],
+                  "dut_host": duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']}
+    if scenario == "Cleanup":
+        ptf_params["routes_removed"] = True
+    
+    if scenario == "Cleanup" and not vc.CLEANUP:
+        logger.info("Skipping cleanup")
+        pytest.skip("Skip cleanup specified")
```

Comment:
I'm not sure I understand what this is doing - is Cleanup a test case, or is it telling the PTF to do some cleanup after the tests are over? Is there any chance that this case fails? If it does, what do we need to do to return the device to the pre-test state?

Reply:
Cleanup is not a test. It is a directive to stop the test without doing cleanup in DUT so we can run warmboot etc (manually) with the configs. If it fails for some reason, it is intended to be manually intervened.

Reply:
I guess my question really is: what happens when `scenario == "Cleanup"` and `vc.CLEANUP` is true?

Reply:
In the cleanup scenario, if `vc.CLEANUP` is `True` then I think the `ptf_runner` should just be running the same tests as the disabled scenario (I'm assuming to check that everything was cleaned up properly). @prsunny Can you confirm?

Reply:
Confirmed offline, resolving